client/my-sites/site-settings/wrap-settings-form.jsx

```diff
@@ -44,39 +45,55 @@ const wrapSettingsForm = getFormSettings => SettingsForm => {
 			this.props.replaceFields( getFormSettings( this.props.settings ) );
 		}
 
-		componentWillReceiveProps( nextProps ) {
-			if ( nextProps.siteId !== this.props.siteId ) {
-				nextProps.clearDirtyFields();
+		componentDidUpdate( prevProps ) {
+			if ( prevProps.siteId !== this.props.siteId ) {
+				this.props.clearDirtyFields();
 			}
 
-			if ( nextProps.settings !== this.props.settings ) {
-				let newState = getFormSettings( nextProps.settings );
-				//If we have any fields that the user has updated,
-				//do not wipe out those fields from the poll update.
-				newState = omit( newState, nextProps.dirtyFields );
-				nextProps.replaceFields( newState );
+			if (
+				! shallowEqual( prevProps.settings, this.props.settings ) ||
+				! shallowEqual( prevProps.fields, this.props.fields )
+			) {
+				this.updateDirtyFields();
 			}
 
 			if (
-				this.props.isSavingSettings &&
-				! nextProps.isSavingSettings
+				! this.props.isSavingSettings &&
+				prevProps.isSavingSettings
 			) {
-				if ( nextProps.isSaveRequestSuccessful ) {
-					nextProps.successNotice( nextProps.translate( 'Settings saved!' ), { id: 'site-settings-save' } );
-					nextProps.clearDirtyFields();
-					nextProps.markSaved();
+				if ( this.props.isSaveRequestSuccessful ) {
+					this.props.successNotice( this.props.translate( 'Settings saved!' ), { id: 'site-settings-save' } );
 				} else {
-					let text = nextProps.translate( 'There was a problem saving your changes. Please try again.' );
-					switch ( nextProps.siteSettingsSaveError ) {
+					let text = this.props.translate( 'There was a problem saving your changes. Please try again.' );
+					switch ( this.props.siteSettingsSaveError ) {
 						case 'invalid_ip':
-							text = nextProps.translate( 'One of your IP Addresses was invalid. Please try again.' );
+							text = this.props.translate( 'One of your IP Addresses was invalid. Please try again.' );
 							break;
 					}
-					nextProps.errorNotice( text, { id: 'site-settings-save' } );
+					this.props.errorNotice( text, { id: 'site-settings-save' } );
 				}
 			}
 		}
 
+		updateDirtyFields() {
+			const unpersistedFields = this.props.fields;
```

Comment:
Still wondering if another name would be more appropriate, since this does include a mix of both persisted and unpersisted fields, yeah? Maybe `allFields` or `currentFields`?